docs/notifications.md

```diff
@@ -67,40 +67,121 @@ envelope. Each event has a unique id that can be used to uniquely identify incom
 requests, if required. Along with that, an _action_ is provided with a
 _target, identifying the object mutated during the event.
 
-The fields available in an event are described in detail in the
-[godoc](http://godoc.org/github.com/docker/distribution/notifications#Event).
+The fields available in an event are described as below
 
-**TODO:** Let's break out the fields here rather than rely on the godoc.
+```
+type Event struct {
+    // ID provides a unique identifier for the event.
+    ID  string `json:"id,omitempty"`
+
+    // Timestamp is the time at which the event occurred.
+    Timestamp time.Time `json:"timestamp,omitempty"`
+
+    // Action indicates what action encompasses the provided event.
+    Action string `json:"action,omitempty"`
+
+    // Target uniquely describes the target of the event.
+    Target struct {
+        distribution.Descriptor
+
+        // Length in bytes of content. Same as Size field in Descriptor.
+        // Provided for backwards compatibility.
+        Length int64 `json:"length,omitempty"`
+
+        // Repository identifies the named repository.
+        Repository string `json:"repository,omitempty"`
+
+        // FromRepository identifies the named repository which a blob was mounted
+        // from if appropriate.
+        FromRepository string `json:"fromRepository,omitempty"`
+
+        // URL provides a direct link to the content.
+        URL string `json:"url,omitempty"`
+
+        // Tag identifies a tag name in tag events
+        Tag string `json:"tag,omitempty"`
+    }   `json:"target,omitempty"`
+
+    // Request covers the request that generated the event.
+    Request RequestRecord `json:"request,omitempty"`
+
+    // Actor specifies the agent that initiated the event. For most
+    // situations, this could be from the authorizaton context of the request.
+    Actor ActorRecord `json:"actor,omitempty"`
+
+    // Source identifies the registry node that generated the event. Put
+    // differently, while the actor "initiates" the event, the source
+    // "generates" it.
+    Source SourceRecord `json:"source,omitempty"`
+}
+```
```

Comment:
I think the intention is for go developers to read this.  

Given that there are JSON examples already (and JSON doesn't support commenting), how about I just expand the object-like sections.  Or put it in a table?